File: src/main/java/main/subReq/allRequirements/CopyReqFromAnotherTender.java
package main.subReq.allRequirements;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class CopyReqFromAnotherTender {
    private WebDriver driver;
    private By tenderToCopyReqFrom = By.xpath("//td[contains(., 'Tender - Copy requirement (for Auto TestProject)')]");
    private By checkBoxOfDataReqToCopy = By.xpath("//input[@type='checkbox' and @name='4575']");
    private By copyReqFromAnotherTenderNameInput = By.xpath("//input[@type='text' and @value='data input test']");
    private By copyButton = By.xpath("//button[text()='Copy']");

    public CopyReqFromAnotherTender(WebDriver driver) {
        this.driver = driver;
    }

    public void selectTenderToCopyReqFrom() {
        WebElement tenderToCopyReqFromElement = driver.findElement(tenderToCopyReqFrom);
        tenderToCopyReqFromElement.click();
    } public void checkBoxOfDataReqToCopy() {
        WebElement checkBoxOfDataReqToCopyElement = driver.findElement(checkBoxOfDataReqToCopy);
        checkBoxOfDataReqToCopyElement.click();
    }

    public void fillIncopyReqFromAnotherTenderNameInput(String newName) {
        WebElement copyReqFromAnotherTenderNameInputElement = driver.findElement(copyReqFromAnotherTenderNameInput);
        copyReqFromAnotherTenderNameInputElement.clear();
        copyReqFromAnotherTenderNameInputElement.sendKeys(newName);
    }

    public void clickCopyButton() {
        WebElement copyButtonElement = driver.findElement(copyButton);
        copyButtonElement.click();
    }
}

File: src/main/java/main/subReq/allRequirements/CopyReqFromAnotherTender.kt
package main.subReq.allRequirements

import org.openqa.selenium.By
import org.openqa.selenium.WebDriver

class CopyReqFromAnotherTender(private val driver: WebDriver) {
    private val tenderToCopyReqFrom: By = By.xpath("//td[contains(., 'Tender - Copy requirement (for Auto TestProject)')]")
    private val checkBoxOfDataReqToCopy: By = By.xpath("//input[@type='checkbox' and @name='4575']")
    private val copyReqFromAnotherTenderNameInput: By = By.xpath("//input[@type='text' and @value='data input test']")
    private val copyButton: By = By.xpath("//button[text()='Copy']")

    fun selectTenderToCopyReqFrom() {
        val tenderToCopyReqFromElement = driver.findElement(tenderToCopyReqFrom)
        tenderToCopyReqFromElement.click()
    }

    fun checkBoxOfDataReqToCopy() {
        val checkBoxOfDataReqToCopyElement = driver.findElement(checkBoxOfDataReqToCopy)
        checkBoxOfDataReqToCopyElement.click()
    }

    fun fillIncopyReqFromAnotherTenderNameInput(newName: String?) {
        val copyReqFromAnotherTenderNameInputElement = driver.findElement(copyReqFromAnotherTenderNameInput)
        copyReqFromAnotherTenderNameInputElement.clear()
        copyReqFromAnotherTenderNameInputElement.sendKeys(newName)
    }

    fun clickCopyButton() {
        val copyButtonElement = driver.findElement(copyButton)
        copyButtonElement.click()
    }
}

